### PR TITLE
fix: Infinite loading horizontal feeds

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

Fixes a situation where scrolling to the end of a horizontal list would fire the `fetchMore` resolver repeatedly. This issue was caused on:  https://github.com/apollographql/apollo-client/issues/6354#issuecomment-664879778

### How do I test this PR?

I tested by copying these changes over to Newspring. It's tricky to reproduce in core, since we don't have any items with children that paginate.

### CODE REVIEW

File: packages/apollos-ui-connected/src/HorizontalFeedConnected.js [hunk 1]
```diff
@@ -122,6 +121,7 @@ class HorizontalFeedConnected extends Component {
         query={this.props.query}
         variables={this.props.variables}
         fetchPolicy={'cache-and-network'}
+        notifyOnNetworkStatusChange
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/HorizontalFeedConnected.js:124]: what is this doing? from the docs, it seems this is only useful if you are using `networkStatus` from the query prop bag, like if you wanted to do something during a refetch

Comment (vinnyjth) [reply packages/apollos-ui-connected/src/HorizontalFeedConnected.js:124]: It also lets you use the `loading` state during a refetch. Normally the loading doesn't flip from `false` -> `true` during a refetch as of apollos 3

Comment (vinnyjth) [reply packages/apollos-ui-connected/src/HorizontalFeedConnected.js:124]: Clue here: https://github.com/apollographql/apollo-client/issues/6354#issuecomment-663584372

Comment (redreceipt) [reply packages/apollos-ui-connected/src/HorizontalFeedConnected.js:124]: oh interesting

**Review by redreceipt — APPROVED** [review]

**Pull request merged**